qiskit/algorithms/optimizers/optimizer.py

```diff
@@ -166,11 +167,55 @@ class OptimizerSupportLevel(IntEnum):
     required = 3  # Feature is required and must be given, None is invalid
 
 
+class OptimizerCallback(Protocol):
```

Comment:
How the optimizers define the callback signature is currently very individual. It might well be that a new optimizer we add (like CMA-ES) will have another signature again. To avoid that we have to change the `OptimizerCallback` every time, would it make sense to just have 
```
__call__(self, x, *args, **kwargs)
```
as signature? That would cover the case that you need, right?

Reply:
I think diversifying in the first place is too complicated for users who write callbacks. Couldn't all cases be covered by the following?
```python
__call__(xk, state: OptimizeResult)
```
It's possible because `OptimizeResult` is a subclass of `dict`.